src/shared/util/dom.tsx

```diff
@@ -12,3 +12,31 @@ export function insertAfter(el: HTMLElement, referenceNode: Node): void {
 export function isMouseEventWithModifierKey(e: MouseEvent): boolean {
     return e.altKey || e.shiftKey || e.ctrlKey || e.metaKey || e.which === 2
 }
+
+export function commitIDFromPermalink({
+    selector,
+    hrefRegex,
+}: {
+    selector: string
+    hrefRegex: RegExp
+}): string | Error {
+    const permalinkElement = document.querySelector(selector)
```

Comment:
Based on this functons name, I would consider casting this to a link and then below we don't have to use `getAttribute`. 

```suggestion
    const permalinkElement = document.querySelector<HTMLAnchorElement>(selector)
```